### PR TITLE
Spatial_oM: Migrate shape profiles from Geomatry_oM to  Spatial_oM 

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes first half of  #756

Migrating all shape profiles from Geometry_oM to Spatial_oM
Adding descriptions to most classes
Aligning affected oMs to the change

### Test files

Versioning test file. Note that this requires to rebuild versioning toolkit for it to work.
https://burohappold.sharepoint.com/:f:/s/BHoM/EsQduRrVa-VFh6nxC5NPzqIB8n8HbzDXB1XGeRw5OToVzQ?e=HlAOxH

### Additional comments

If done early enough, I do not mind any comments on the descriptions, but once all dependant PRs are raised, think we are better of merging with typo than holding up all PRs to come!

### CODE REVIEW

File: Spatial_oM/ShapeProfiles/ChannelProfile.cs [hunk 1]
```diff
@@ -36,20 +39,34 @@ public class ChannelProfile : BHoMObject, IProfile, IImmutable
         /***************************************************/
         public virtual ShapeType Shape { get; } = ShapeType.Channel;
 
+        [Length]
+        [Description("Full depth of the profile from bottom of the bottom flange to top of the top flange.")]
         public virtual double Height { get; }
 
+        [Length]
+        [Description("Full width of the profile from the furthest side of the web to the end of the flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ChannelProfile.cs:47]: Full width between the extreme fibres of the web to the flange.

File: Spatial_oM/ShapeProfiles/ChannelProfile.cs [hunk 2]
```diff
@@ -36,20 +39,34 @@ public class ChannelProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double FlangeWidth { get; }
 
+        [Length]
+        [Description("Thickness of the stem of the C shape.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ChannelProfile.cs:51]: Not needed.

File: Spatial_oM/ShapeProfiles/ChannelProfile.cs [hunk 3]
```diff
@@ -36,20 +39,34 @@ public class ChannelProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double WebThickness { get; }
 
+        [Length]
+        [Description("Thickness of the top and base of the C shape.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ChannelProfile.cs:55]: Not needed.

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs [hunk 4]
```diff
@@ -25,9 +25,12 @@
 using System.Collections.ObjectModel;
 using System.Collections.Generic;
 using System.Linq;
+using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
 
-namespace BH.oM.Geometry.ShapeProfiles
+namespace BH.oM.Spatial.ShapeProfiles
 {
+    [Description("Rectangular hollow profile that allows for varying thickness of the webs and both flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:33]: Rectangular hollow profile that allows for different thicknesses of the webs, top flange and bottom flange.

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs [hunk 5]
```diff
@@ -36,18 +39,29 @@ public class FabricatedBoxProfile : BHoMObject, IProfile, IImmutable
         /***************************************************/
         public virtual ShapeType Shape { get; } = ShapeType.Box;
 
+        [Length]
+        [Description("Full depth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:43]: Full depth between the extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs [hunk 6]
```diff
@@ -36,18 +39,29 @@ public class FabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Height { get; }
 
+        [Length]
+        [Description("Full width.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:47]: Full width between the extreme fibres of the webs.

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs [hunk 7]
```diff
@@ -36,18 +39,29 @@ public class FabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Width { get; }
 
+        [Length]
+        [Description("Thickness of both webs, i.e. both of the sides.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:51]: Not required, clarified in the description.

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs [hunk 8]
```diff
@@ -36,18 +39,29 @@ public class FabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double WebThickness { get; }
 
+        [Length]
         public virtual double TopFlangeThickness { get; }
 
+        [Length]
         public virtual double BotFlangeThickness { get; }
 
+        [Length]
+        [Description("Fillet weld size between inside of webs and flanges. Measured as the distance between intersection of web and flange perpendicular to the edge of the weld.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:61]: Not sure you can have fillet welds on the inside of a box. You wouldn't be able to seal the box if it was:
image

"Fillet weld size between the webs and flanges. Measured as the throat thickness of the weld."

If throat thickness is too structural, "the perpendicular distance from the corner of the web to it's longest side."

File: Spatial_oM/ShapeProfiles/FabricatedISectionProfile.cs [hunk 9]
```diff
@@ -25,9 +25,12 @@
 using System.Collections.ObjectModel;
 using System.Collections.Generic;
 using System.Linq;
+using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
 
-namespace BH.oM.Geometry.ShapeProfiles
+namespace BH.oM.Spatial.ShapeProfiles
 {
+    [Description("I-shaped profile with parallel flanges that allows for varying thicknesses of webs and flanges as well as different top and bottom flange widths.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedISectionProfile.cs:33]: "...allows for different thicknesses of the web, top flange and bottom flange as well as different top and bottom flange widths."

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 10]
```diff
@@ -25,9 +25,12 @@
 using System.Linq;
 using BH.oM.Base;
 using BH.oM.Geometry;
+using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
 
-namespace BH.oM.Geometry.ShapeProfiles
+namespace BH.oM.Spatial.ShapeProfiles
 {
+    [Description("Rectangular hollow profile that allows for varying thickness of the webs and both flanges as well as freely defined outstands of the top and bottom flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:33]: "...that allows for different thicknesses of the webs, top flange and bottom flange as well as defined outstands of the top and bottom flange."

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 11]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
 
         public virtual ShapeType Shape { get; } = ShapeType.Box;
 
+        [Length]
+        [Description("Full depth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:43]: Full depth between the extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 12]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Height { get; }
 
+        [Length]
+        [Description("Full width.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:47]: Full width between the extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 13]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Width { get; }
 
+        [Length]
+        [Description("Thickness of both webs, i.e. both of the sides.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:51]: Not needed, clarified in the description.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 14]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double WebThickness { get; }
 
+        [Length]
         public virtual double TopFlangeThickness { get; }
 
+        [Length]
         public virtual double BotFlangeThickness { get; }
 
+        [Length]
+        [Description("Outstand along the top flange in the top left corner of the profile.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:61]: The additional width added to the left side of the top flange.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 15]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double TopLeftCorbelWidth { get; }
 
+        [Length]
+        [Description("Outstand along the top flange in the top right corner of the profile.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:65]: The additional width added to the right side of the top flange.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 16]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double TopRightCorbelWidth { get; }
 
+        [Length]
+        [Description("Outstand along the bottom flange in the bottom left corner of the profile.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:69]: The additional width added to the left side of the bottom flange.

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 17]
```diff
@@ -36,24 +39,41 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double BotLeftCorbelWidth { get; }
 
+        [Length]
+        [Description("Outstand along the bottom flange in the bottom right corner of the profile.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs:73]: The additional width added to the right side of the bottom flange.

File: Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs [hunk 18]
```diff
@@ -25,9 +25,12 @@
 using System.Linq;
 using BH.oM.Base;
 using BH.oM.Geometry;
+using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
 
-namespace BH.oM.Geometry.ShapeProfiles
+namespace BH.oM.Spatial.ShapeProfiles
 {
+    [Description("T-shaped profile that allows for varying outstand widths as well as varying outstand thicknesses. Outstands width varying thickness are aligned at the top.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs:33]: "..allows for different outsand widths and thicknesses. Outstands with different thicknesses are aligned by their top edge. The full width of the top flange can be calculated as LeftOutstandWidth + WebThickness + RightOutstandWidth."

File: Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs [hunk 19]
```diff
@@ -36,20 +39,32 @@ public class GeneralisedTSectionProfile : BHoMObject, IProfile, IImmutable
 
         public virtual ShapeType Shape { get; } = ShapeType.Tee;
 
+        [Length]
+        [Description("Full depth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs:43]: Full depth between the extreme fibre of the outstand and the web.

File: Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs [hunk 20]
```diff
@@ -36,20 +39,32 @@ public class GeneralisedTSectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Height { get; }
 
+        [Length]
         public virtual double WebThickness { get; }
 
+        [Length]
+        [Description("Outstand width of the top left corner. Measured as the distance from the left side of the web to the end of the outstand.\n"
+                    + "The full width ot the top flange can be calculated as LeftOutstandWidth + WebThickness + RightOutstandWidth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs:51]: Outstand width on the left side, measured from the outside edge of the web.

File: Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs [hunk 21]
```diff
@@ -36,20 +39,32 @@ public class GeneralisedTSectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double LeftOutstandWidth { get; }
 
+        [Length]
         public virtual double LeftOutstandThickness { get; }
 
+        [Length]
+        [Description("Outstand width of the top right corner. Measured as the distance from the right side of the web to the end of the outstand.\n"
+                    + "The full width ot the top flange can be calculated as LeftOutstandWidth + WebThickness + RightOutstandWidth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedTSectionProfile.cs:59]: Outstand width on the right side, measured from the outside edge of the web.

File: Spatial_oM/ShapeProfiles/ISectionProfile.cs [hunk 22]
```diff
@@ -37,18 +40,30 @@ public class ISectionProfile : BHoMObject, IProfile, IImmutable
 
         public virtual ShapeType Shape { get; } = ShapeType.ISection;
 
+        [Length]
+        [Description("Full depth of the profile from bottom of the bottom flange to top of the top flange.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ISectionProfile.cs:44]: Full depth between the extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/ISectionProfile.cs [hunk 23]
```diff
@@ -37,18 +40,30 @@ public class ISectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Height { get; }
 
+        [Length]
+        [Description("Full width both top and bottom flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ISectionProfile.cs:48]: Full width between the extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/ISectionProfile.cs [hunk 24]
```diff
@@ -37,18 +40,30 @@ public class ISectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Width { get; }
 
+        [Length]
         public virtual double WebThickness { get; }
 
+        [Length]
+        [Description("Thickness of both flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/ISectionProfile.cs:55]: Not needed.

File: Spatial_oM/ShapeProfiles/TSectionProfile.cs [hunk 25]
```diff
@@ -36,20 +39,34 @@ public class TSectionProfile : BHoMObject, IProfile, IImmutable
         /***************************************************/
         public virtual ShapeType Shape { get; } = ShapeType.Tee;
 
+        [Length]
+        [Description("Full depth.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/TSectionProfile.cs:43]: Full depth between extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/TSectionProfile.cs [hunk 26]
```diff
@@ -36,20 +39,34 @@ public class TSectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Height { get; }
 
+        [Length]
+        [Description("Full width of the flange.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/TSectionProfile.cs:47]: Full width between extreme fibres of the flanges.

File: Spatial_oM/ShapeProfiles/TSectionProfile.cs [hunk 27]
```diff
@@ -36,20 +39,34 @@ public class TSectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double Width { get; }
 
+        [Length]
+        [Description("Thickness of the stem of the T-shape.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/TSectionProfile.cs:51]: Not needed

File: Spatial_oM/ShapeProfiles/TSectionProfile.cs [hunk 28]
```diff
@@ -36,20 +39,34 @@ public class TSectionProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double WebThickness { get; }
 
+        [Length]
+        [Description("Thickness of the top of the T-shape")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/TSectionProfile.cs:55]: Not needed

File: Spatial_oM/ShapeProfiles/AngleProfile.cs [hunk 29]
```diff
@@ -36,22 +39,37 @@ public class AngleProfile : BHoMObject, IProfile, IImmutable
         /***************************************************/
         public virtual ShapeType Shape { get; } = ShapeType.Angle;
 
+        [Length]
+        [Description("Full depth of the profile from bottom of flange to top of the web.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/AngleProfile.cs:43]: Full depth between the extreme fibres of the flange and web.

**Review by peterjamesnugent — COMMENTED** [review]
A few simplifications of descriptions and removal of descriptions where they can be clarified in the description.

We need a better way of describing "bottom of the bottom flange to top of the top flange" - I have opted for using "extreme fibres" but as discussed this might be too structural.

File: Spatial_oM/ShapeProfiles/AngleProfile.cs [hunk 30]
```diff
@@ -36,22 +39,37 @@ public class AngleProfile : BHoMObject, IProfile, IImmutable
...
+        [Length]
+        [Description("Thickness of the base of the L-shape.")]
         public virtual double FlangeThickness { get; }
 
+        [Length]
+        [Description("Fillet radius between inner face of the flange and inner face of the web.")]
         public virtual double RootRadius { get; }
 
+        [Length]
+        [Description("Fillet radius at the outer edge of the flange and web. Value need to be smaller or equal than the Flange and Web thicknesses")]
```
[the remark this answers is not in the document]
Comment (IsakNaslundBh) [reply Spatial_oM/ShapeProfiles/AngleProfile.cs:63]: it is a different radius though. It is not between but at, but have reworded slightly

File: Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs (hunk 8 above)
Comment (IsakNaslundBh) [reply Spatial_oM/ShapeProfiles/FabricatedBoxProfile.cs:61]: As discussed, keeping current description for now, as that is actually what it is doing. TO be resolved in another issue

File: Spatial_oM/ShapeProfiles/ISectionProfile.cs (hunk 24 above)
Comment (IsakNaslundBh) [reply Spatial_oM/ShapeProfiles/ISectionProfile.cs:55]: keeping as the "both" words adds some information

File: Spatial_oM/ShapeProfiles/AngleProfile.cs [hunk 31]
```diff
@@ -40,33 +40,31 @@ public class AngleProfile : BHoMObject, IProfile, IImmutable
...
-        [Description("Thickness of the base of the L-shape.")]
         public virtual double FlangeThickness { get; }
 
         [Length]
         [Description("Fillet radius between inner face of the flange and inner face of the web.")]
         public virtual double RootRadius { get; }
 
         [Length]
-        [Description("Fillet radius at the outer edge of the flange and web. Value need to be smaller or equal than the Flange and Web thicknesses")]
+        [Description("Fillet radius at the end of the flange and the end of the web. Value need to be smaller or equal than the Flange and Web thicknesses")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/AngleProfile.cs:61]: flange
web

File: Spatial_oM/ShapeProfiles/FabricatedISectionProfile.cs [hunk 32]
```diff
@@ -40,7 +40,7 @@ public class FabricatedISectionProfile : BHoMObject, IProfile, IImmutable
         public virtual ShapeType Shape { get; } = ShapeType.ISection;
 
         [Length]
-        [Description("Full depth.")]
+        [Description("Full width between the extreme fibres of the flanges.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/FabricatedISectionProfile.cs:43]: Full depth

File: Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs [hunk 33]
```diff
@@ -58,19 +57,19 @@ public class GeneralisedFabricatedBoxProfile : BHoMObject, IProfile, IImmutable
...
         public virtual double TopRightCorbelWidth { get; }
 
         [Length]
-        [Description("Outstand along the bottom flange in the bottom left corner of the profile.")]
+        [Description("The additional width added to the left side of the bottom flange.")]
         public virtual double BotLeftCorbelWidth { get; }
 
         [Length]
-        [Description("Outstand along the bottom flange in the bottom right corner of the profile.")]
+        [Description("The additional width added to the right side of the bottom flange.")]
```
Comment (peterjamesnugent) [inline Spatial_oM/ShapeProfiles/GeneralisedFabricatedBoxProfile.cs]: In each of these we should mention:
"..., measure from the outside edge of the web."

**Review by peterjamesnugent — COMMENTED** [review]
Last few changes.

**Review by peterjamesnugent — APPROVED** [review]
LGTM.

**Pull request merged**